src/actions/index.js

```diff
@@ -69,14 +84,14 @@ const apiAction = (method, dataKey, options) => _dispatch => {
 
             // Handle the case that calling response.json() on null responses throws a syntax error
             const text = await response.text()
-            const json = text ? JSON.parse(text) : {}
+            const data = getDataFromResponseText({ text })
 
             // Handle any request errors since fetch doesn't throw
             if (!response.ok) {
                 const { status, statusText } = response
                 throw new ErrorClass(status, statusText, {
```

Comment:
@kourge i think this will surface more accurate error from api response, instead of generic json syntax error

Reply:
Yes, you're right, thanks for pointing out that it does get logged. I totally missed it.